### PR TITLE
TST: parametrize and de-duplicate arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is likely to be a sisyphean task...

### CODE REVIEW

File: pandas/tests/arithmetic/test_datetime64.py [hunk 1]
```diff
@@ -994,10 +999,6 @@ def test_dti_add_timestamp_raises(self, box):
         msg = "cannot add"
         with tm.assert_raises_regex(TypeError, msg):
             idx + Timestamp('2011-01-01')
-
-    def test_dti_radd_timestamp_raises(self):
-        idx = DatetimeIndex(['2011-01-01', '2011-01-02'])
-        msg = "cannot add DatetimeIndex and Timestamp"
         with tm.assert_raises_regex(TypeError, msg):
             Timestamp('2011-01-01') + idx
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_datetime64.py]: Decided that separating tests for `__add__` vs `__radd__` was overkill.

**Comment by jreback** [comment]
looks ok. can you merge master, and does this need to go before #22862 ?

**Comment by jbrockmendel** [comment]
>can you merge master, and does this need to go before #22862 ?

Just pushed.  22862 is PeriodArray, maybe you meant #23215?  Should be fine either way.

**Comment by jreback** [comment]
thanks!

**Pull request merged**